services/src/pro/datasets/in_memory.rs

```diff
@@ -372,7 +471,10 @@ impl
         >,
     > {
         ensure!(
-            self.dataset_permissions
+            self.backend
+                .read()
+                .await
+                .dataset_permissions
```

Comment:
and here

Reply:
fccf47a2459c22ceb51d2ac1d1b4e0237d13cfd1